api/models/GameStateRow.js

```diff
@@ -132,6 +132,11 @@ module.exports = {
       type: 'string',
       allowNull: true,
     },
+    oneOffTargetType: {
+      type: 'string',
+      allowNull: true,
+      isIn: ['point', 'jack', 'faceCard'],
+    },
```

Comment:
Need migration to support this

Reply:
```
create type target_type as enum ('point', 'faceCard', 'jack');

alter table gamestaterow 
add column "oneOffTargetType" target_type;
```

Reply:
Actually it turns out that `text[]` columns aren't supported, so I'm updating the column types to use jsonb instead. It should be it's own PR but for ease's sake I'm including it here. Need this migration to recreate the table:
```
drop table if exists gamestaterow;

CREATE TABLE gamestaterow (
    id SERIAL PRIMARY KEY,
    "gameId" INTEGER NOT NULL REFERENCES game(id) ON DELETE CASCADE,
    "playedBy" SMALLINT NOT NULL CHECK ("playedBy" IN (0, 1)),
    "moveType" TEXT NOT NULL,
    "playedCard" TEXT,
    "targetCard" TEXT,
    "discardedCards" JSONB DEFAULT '[]'::JSONB,
    turn INTEGER NOT NULL,
    phase SMALLINT NOT NULL CHECK (phase IN (1, 2, 3, 4, 5, 7)),
    "p0Hand" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1Hand" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p0Points" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1Points" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p0FaceCards" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1FaceCards" JSONB NOT NULL DEFAULT '[]'::JSONB,
    deck JSONB NOT NULL DEFAULT '[]'::JSONB,
    scrap JSONB NOT NULL DEFAULT '[]'::JSONB,
    "oneOff" TEXT,
    "oneOffTarget" TEXT,
    twos JSONB NOT NULL DEFAULT '[]'::JSONB,
    resolved TEXT,
    "createdAt" TIMESTAMPTZ DEFAULT NOW(),
    "updatedAt" TIMESTAMPTZ DEFAULT NOW()
);


-- Indexes for foreign key fields
CREATE INDEX idx_gameId ON gamestaterow("gameId");
```

Reply:
NOPE THAT'S WRONG it should be

```
drop table if exists gamestaterow;
create type target_type as enum('point', 'faceCard', 'jack');



CREATE TABLE gamestaterow (
    id SERIAL PRIMARY KEY,
    "gameId" INTEGER NOT NULL REFERENCES game(id) ON DELETE CASCADE,
    "playedBy" SMALLINT NOT NULL CHECK ("playedBy" IN (0, 1)),
    "moveType" TEXT NOT NULL,
    "playedCard" TEXT,
    "targetCard" TEXT,
    "discardedCards" JSONB DEFAULT '[]'::JSONB,
    turn INTEGER NOT NULL,
    phase SMALLINT NOT NULL CHECK (phase IN (1, 2, 3, 4, 5, 7)),
    "p0Hand" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1Hand" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p0Points" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1Points" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p0FaceCards" JSONB NOT NULL DEFAULT '[]'::JSONB,
    "p1FaceCards" JSONB NOT NULL DEFAULT '[]'::JSONB,
    deck JSONB NOT NULL DEFAULT '[]'::JSONB,
    scrap JSONB NOT NULL DEFAULT '[]'::JSONB,
    "oneOff" TEXT,
    "oneOffTarget" TEXT,
    "oneOffTargetType" target_type,
    twos JSONB NOT NULL DEFAULT '[]'::JSONB,
    resolved TEXT,
    "createdAt" TIMESTAMPTZ DEFAULT NOW(),
    "updatedAt" TIMESTAMPTZ DEFAULT NOW()
);


-- Indexes for foreign key fields
CREATE INDEX idx_gameId ON gamestaterow("gameId");
```